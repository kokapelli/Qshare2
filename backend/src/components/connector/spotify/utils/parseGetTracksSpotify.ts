import { QShareTrack, QShareArtist } from "../../../music/musicTypes";

interface SpotifyTrack {
    name: string;
    uri: string;
    artists: SpotifyArtist[];
    album: SpotifyAlbum;
    popularity: number;
}

interface SpotifyArtist {
    uri: string;
    name: string;
}

interface SpotifyAlbum {
    images: SpotifyAlbumImage[];
}

interface SpotifyAlbumImage {
    width: number;
    height: number;
    url: string;
}

interface SpotifyGetTracksResponse {
    tracks: {
        name: string
        items: SpotifyTrack[]
        limit: number
    };
}

export interface QShareGetTrackResponse extends QShareTrack {
    popularity: number;
}

export interface QShareGetTracksResponse {
    limit: number;
    tracks: QShareGetTrackResponse[];
}

export const parseGetTracksSpotify = (data: SpotifyGetTracksResponse) => {
    const response: QShareGetTracksResponse = {} as any;
    const tracks: QShareGetTrackResponse[] = [];

    response.tracks = tracks;
    response.limit = data.tracks.limit;
    for (const track of Array.from(data.tracks.items)) {
        const newTrack: QShareGetTrackResponse = {} as any;
        newTrack.name = track.name;

        const newArtists: QShareArtist[] = [];
        for (const artist of track.artists) {
            const newArtist: SpotifyArtist = {
                uri: artist.uri,
                name: artist.name
            };

            newArtists.push(newArtist);
        }

        newTrack.artists = newArtists;

        const newImages = [];
        for (const image of track.album.images) {
            const newImage: SpotifyAlbumImage = {
                height: image.height,
                width: image.width,
                url: image.url
            };
            newImages.push(newImage);
        }

        newTrack.images = newImages;

        newTrack.uri = track.uri;
        newTrack.popularity = track.popularity;
        tracks.push(newTrack);
    }

    return response;
};
