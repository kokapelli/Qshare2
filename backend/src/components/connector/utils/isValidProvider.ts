import { SPOTIFY_PROVIDER } from "../../constants/providers";

const isValidProvider = (provider: string) => {
    return provider === SPOTIFY_PROVIDER;
};

export { isValidProvider };