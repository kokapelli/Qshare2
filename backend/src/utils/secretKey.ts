import crypto from "crypto";

const SECRET_KEY = crypto.randomBytes(256);

export default SECRET_KEY;
