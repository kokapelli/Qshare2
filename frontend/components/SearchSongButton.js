import React, {Component}  from 'react';
import {connect} from 'react-redux';
import { TouchableOpacity, Text, StyleSheet, View, } from 'react-native';
import { Icon } from 'expo';
import _ from 'lodash';
import {addSong} from '../actions/queue'

export class SearchSongButton extends Component
{
  render () {
    const {item, queueID} = this.props;
    if (item && queueID) {
      const {artists} = item;
      const name = _.truncate(item.name, {length:20});
      const artist = artists[0]? _.truncate(artists[0].name, {length:20}):'Undefined';
      return (
        <View style={styles.searchItem}>
          <View style={styles.songName}>
            <Text testID='songDetails' style={styles.songText}>
              {name} - {artist}
            </Text>
          </View>
          <TouchableOpacity 
                testID='addSong' 
                style={styles.addButton} 
                onPress={() => {this.props.addSong(queueID, item)}}>
            <Icon.Ionicons 
              testID='addSongIcon' 
              name={'md-add'} 
              size={50} 
              color="white"/>
          </TouchableOpacity>
        </View>
      )
    }

    return (
      <View style={styles.searchItem}>
        <Text />
      </View>
    )
  }
}

const mapStateToProps = (state) => ({
  queueID: state.queue.queueID
})

export default connect(mapStateToProps, {addSong})(SearchSongButton)

const styles = StyleSheet.create({
  searchItem: {
    height: 80,
    borderWidth: 1,
    paddingLeft: 30,
    paddingRight: 30,
    borderRadius: 50,
    marginTop: 5,
    marginLeft: 10,
    marginRight: 10,
    borderColor: '#383838',
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: '#282828',
  },
  songName: {
    flex: 4,
    justifyContent: 'center',
    backgroundColor: '#282828',
    zIndex: 2
  },
  songText: {
    color: 'white',
    fontSize: 25,
  },
  addButton: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor:'transparent',
  },
})
