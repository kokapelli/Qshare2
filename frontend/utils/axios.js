import Axios from "axios"
import { settings } from "../settings/settings";

const axios = Axios.create()
axios.defaults.headers.common['Provider'] = settings.provider

export default axios