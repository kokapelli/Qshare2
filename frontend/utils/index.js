export const findByTestAttr = (component, attr)  => {
    const wrapper = component.find(`[testID='${attr}']`);
    return wrapper
  }
  