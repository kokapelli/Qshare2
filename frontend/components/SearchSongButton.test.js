import { SearchSongButton } from './SearchSongButton'
import { shallow, configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { findByTestAttr } from '../utils/'

configure({ adapter: new Adapter() });

const item = {
    name: "Qshare",
    artists: [
        {
            name: "Hampus"
        },
        {
            name: "Brent"
        },
        {
            name: "Valter"
        }
    ]
}

describe('SearchSongButton Snapshot', () => {

    it('Matches the snapshot', () => {
        const tree = renderer.create(
        <SearchSongButton />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('SearchSongButton Units', () => {

    beforeEach(()=>{
       this.component = shallow(<SearchSongButton/>)
    })

    it('Component renders properly', () => {
        const rendered = renderer.create(<SearchSongButton/>).toJSON()
        expect(rendered).toBeTruthy()
    });

    it('Ddd song button should be clickable', () => {
        this.component.setProps({ item: item, queueID: "Test queueID" });
        const touchable = findByTestAttr(this.component, 'addSong')
        expect(touchable.exists()).toEqual(true) 
        expect(touchable.simulate('click')).toBeDefined()
        expect(touchable.simulate('click')).toEqual({})
        })

    it('Song Details should be correct given mock input', () => {
        this.component.setProps({ item: item, queueID: "Test queueID" });
        const details = findByTestAttr(this.component, 'songDetails')
        const expected = [item.name, " - ", item.artists[0].name]
        expect(details.props().children).toEqual(expected)
        })

    it('Add Song Icon should render', () => {
        const icon = findByTestAttr(this.component, 'songAddIcon')
        expect(icon).toBeDefined()
    });
})


