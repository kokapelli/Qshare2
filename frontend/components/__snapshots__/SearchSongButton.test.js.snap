// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSongButton Snapshot Matches the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#282828",
      "borderColor": "#383838",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "paddingLeft": 30,
      "paddingRight": 30,
    }
  }
>
  <Text />
</View>
`;
